EventSourcing.Example.Tests/TestsBase.cs

```diff
@@ -0,0 +1,27 @@
+using System.Reflection;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace EventSourcing.Example.Tests;
+
+public class TestsBase : IAsyncLifetime
+{
+  // Control the number of concurrent Tests
+  private static readonly SemaphoreSlim Semaphore = new(8);
+  public async Task InitializeAsync() => await Semaphore.WaitAsync();
+  public Task DisposeAsync() => Task.FromResult(Semaphore.Release());
+  private static readonly TestServer Server = GetServer();
+  protected static TService? GetService<TService>() => GetServer().Services.GetService<TService>();
+  protected static TestServer GetServer()
```

Comment:
You can (and should!) use the same server for all tests! This will speed things up for sure!